kotlinx-coroutines-core/common/src/flow/operators/Zip.kt

```diff
@@ -137,65 +238,52 @@ public inline fun <T1, T2, T3, T4, T5, R> Flow<T1>.combineLatest(
  * Returns a [Flow] whose values are generated with [transform] function by combining
  * the most recently emitted values by each flow.
  */
-@FlowPreview
-public inline fun <reified T, R> Flow<T>.combineLatest(vararg others: Flow<T>, crossinline transform: suspend (Array<T>) -> R): Flow<R> =
-    combineLatest(*others, arrayFactory = { arrayOfNulls(others.size + 1) }, transform = { transform(it) })
+@ExperimentalCoroutinesApi
+public inline fun <reified T, R> combine(
+    vararg flows: Flow<T>,
+    crossinline transform: suspend (Array<T>) -> R
+): Flow<R> = combine(*flows, arrayFactory = { arrayOfNulls(flows.size) }, transform = { emit(transform(it)) })
```

Comment:
This `combine` implementation calls array-based `combine` which is actually a combine-transform that is using _safe_ collector that is not needed here.